searchcore/src/vespa/searchcore/proton/matching/query.cpp

```diff
@@ -156,6 +157,11 @@ void
 Query::setWhiteListBlueprint(Blueprint::UP whiteListBlueprint)
 {
     _whiteListBlueprint = std::move(whiteListBlueprint);
+    using proton::documentmetastore::WhiteListProvider;
+    auto wlf = dynamic_cast<WhiteListProvider *>(_whiteListBlueprint.get());
+    if (wlf) {
+        _global_white_list = wlf->get_white_list_filter();
```

Comment:
This adds a cost (creating a copy of the white list bitvector) even when this copy won't be used.

Reply:
Agree with Tor. Can we do this in optimize() instead, where we know we need it? In that case we need to store a pointer to WhiteListProvider in Query as the white list blueprint are moved into the blueprint tree between this call and the call to optimize().